UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs

```diff
@@ -819,17 +834,315 @@ public IEnumerator ProcessDNAPayload(DNAMutationData InDNAMutationData)
+						bodyArmr.Armor.DismembermentProtectionChance = 0;
+						bodyArmr.Armor.StunImmunity = false;
+						bodyArmr.Armor.TemperatureProtectionInK = new Vector2(283.15f, 283.15f + 20);
+						bodyArmr.Armor.PressureProtectionInKpa = new Vector2(30f, 300f);
+					}
+					continue;
+				}
+				if (comp is IDCard card)
+				{
+					var cardJobName = card.GetJobTitle();
+					var cardRegName = card.RegisteredName;
+					card.ServerChangeOccupation(OccupationList.Instance.Get(JobType.ASSISTANT), true, true);
+
+					card.ServerSetRegisteredName(cardRegName);
+					card.ServerSetJobTitle(cardJobName);
+					for (int i = 0; i < card.currencies.Length; i++)
+					{
+						card.currencies[i] = 0; // removing all currencies on card to be sure
+					}
+					continue;
+				}
+				if (comp is ItemStorage itemStorage)
+				{
+					itemStrg = itemStorage;
+					continue;
+				}
+				if (comp is ItemActionButton actionButton)
+				{
+					actionButton.OnRemovedFromBody(this);
+					continue;
+				}
+				if (comp is not Pickupable && comp is not UprightSprites && comp is not UniversalObjectPhysics
+				 && comp is not SortingGroup && comp is MonoBehaviour mono)
+				{
+					mono.NetDisable();
+					continue;
+				}
+			}
+			return itemStrg;
+		}
+
+		private void RemoveItemsInsideFakeItem(GameObject fakeItem)
+		{
+			if (fakeItem.TryGetComponent<InteractableStorage>(out var iS))
+			{
+				var items = iS.ItemStorage.GetItemSlots();
+
+				foreach (var x in items)
+				{
+					if (x.ItemObject != null)
+						_ = Despawn.ServerSingle(x.ItemObject);
+				}
+			}
+		}
+
+		private void SetUpFakeItems(ChangelingDna dna, ChangelingMain changeling)
+		{
+			if (dna != null && changeling != null)
+			{
+				var storage = changeling.ChangelingMind.CurrentPlayScript.DynamicItemStorage;
+				// need to firstly create fake uniform for placing id card into fake slot
+				dna = SortFakeItems(dna);
+
+				foreach (var id in dna.BodyClothesPrefabID)
+				{
+					var fakeItem = Spawn.ServerPrefab(CustomNetworkManager.Instance.ForeverIDLookupSpawnablePrefabs[id]).GameObject;
+					ItemSlot bestSlot = storage.GetBestSlotFor(fakeItem);
+					if (bestSlot == null)
+					{
+						_ = Despawn.ServerSingle(fakeItem);
+						continue;
+					}
+					var placed = Inventory.ServerAdd(fakeItem, bestSlot);
+					// better don`t put fake items into storages
+					if (placed == false || (((int?)bestSlot.NamedSlot) > 15) || bestSlot.NamedSlot == NamedSlot.handcuffs)
+					{
+						_ = Despawn.ServerSingle(fakeItem);
+						continue;
+					}
+					fakeItem.GetComponent<ItemAttributesV2>().IsFakeItem = true;
+					// making items absolutely useless
+					RemoveItemsInsideFakeItem(fakeItem);
+					ItemStorage itemStrg = MakeFakeItemUseless(fakeItem);
+
+
+					var itemName = fakeItem.GetComponent<ItemAttributesV2>().InitialName;
+					// removing item anytime when item was moved or something
+					fakeItem.GetComponent<Pickupable>().OnInventoryMoveServerEvent.AddListener((GameObject item) =>
```

Comment:
Can you ensure that `OnInventoryMoveServerEvent` clears all its listeners when destroyed and that the fake items actually get destroyed? Because this could cause a memory leak due to a lambda function never getting cleared.

Reply:
Yep, OnInventoryMoveServerEvent clear all listeners in Pickupable class when gets despawned by server.